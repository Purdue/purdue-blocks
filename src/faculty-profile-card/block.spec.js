jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    updateRangeInput,
    clickCheckbox,
    clickRadio,
    openSidebarPanelWithTitle
} from '../test-helpers'

const block = {blockTitle: 'Faculty Profile Card', blockName: 'purdue-blocks/faculty-profile-card'}


// cannot use match snapshot as a test case because everytime a new page is created, it will have a new id, 
// and that id is captured in a block attribute and in the snapshot, so it will change every time 
// the test runs.
describe( 'ðŸ”¬ Faculty Profile Card Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock( block.blockTitle )


        // tests that the block is properly inserted
        expect( await getEditedPostContent() ).toMatchSnapshot()
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
    } )

    describe('ðŸ”¬ Block Editor Fields', () => {
        
        test('ðŸ”Ž Media Library Button', async () =>{
            await blockStartup(block)

            // open media library
            await clickElementByText('button', 'Open Media Library')
    
            // tests that media library does open
            expect(await page.$(`div.media-frame-title`)).not.toBeNull()
        })

        describe('ðŸ”¬ Personal Info Fields', () => {
            test('ðŸ”Ž Add Phone Number Field', async () =>{
                await blockStartup(block)
                
                const typeString = "123 456 7890"
        
                await page.focus('input[placeholder="Phone Number..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"phone":"${typeString}"`)).toBe(true)
            })
            test('ðŸ”Ž Add Email Address Field', async () =>{
                await blockStartup(block)
                
                const typeString = "test@test.test"
        
                await page.focus('input[placeholder="Email..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"email":"${typeString}"`)).toBe(true)
            })
            test('ðŸ”Ž Add Personal Website Field', async () =>{
                await blockStartup(block)
                
                const typeString = "https://www.purdue.edu"
        
                await page.focus('input[placeholder="Personal Site..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"personalLink":"${typeString}"`)).toBe(true)
            })
        })
    })

    describe('ðŸ”¬ Side Panel Settings', () => {
        test('ðŸ”Ž Image alt text field', async () => {
            await blockStartup(block)

            const typeString = "Image alt text."
            
            // focus the text box then type into it with the virtual keyboard
            await clickElementByText('label', 'Profile Picture Alt Text')
            await page.keyboard.type(typeString, {delay: 10})
    
            const editedContent = await getEditedPostContent()
    
            expect( editedContent.includes(`"altText":"${typeString}"`)).toBe(true)
            expect( editedContent).toMatchSnapshot()
        })

        describe('ðŸ”¬ Social Share Links', () => {

            // test settings in the sidebar for adding the social share links to the hero
            test( 'ðŸ”Ž "Include Social Share Links" checkbox reveals individual checkboxes', async () => {
                await blockStartup(block)
        
                await clickCheckbox('Include Social Media Links')
        
                const socialsCheckboxes = await page.$x(
                    `//label[@class = "components-checkbox-control__label"][.="Facebook" or .="Twitter" or .="LinkedIn" or .="Instagram"]`
                )
        
                const editedContent = await getEditedPostContent()
                // check that the checkboxes appear and that the correct number exist, 3
                expect(socialsCheckboxes).not.toBeNull()
                expect(socialsCheckboxes.length).toBe(4)
                expect( editedContent ).toMatchSnapshot()
            })

            test( 'ðŸ”Ž Social links added correctly', async () => {
                await blockStartup(block)
                await clickCheckbox('Include Social Media Links')

                await clickCheckbox('Facebook')
                let editedContent = await getEditedPostContent()
                expect( editedContent.includes('"Facebook":{"slug":"facebook-square","checked":true}')).toBe(true)
                
                await clickCheckbox('Twitter')
                editedContent = await getEditedPostContent()
                expect( editedContent.includes('"Twitter":{"slug":"twitter-square","checked":true}')).toBe(true)
                
                await clickCheckbox('LinkedIn')
                editedContent = await getEditedPostContent()
                expect( editedContent.includes('"LinkedIn":{"slug":"linkedin","checked":true}')).toBe(true)
                
                await clickCheckbox('Instagram')
                editedContent = await getEditedPostContent()
                expect( editedContent.includes('"Instagram":{"slug":"instagram","checked":true}')).toBe(true)

                expect( editedContent).toMatchSnapshot()
            })
        })
    })
})