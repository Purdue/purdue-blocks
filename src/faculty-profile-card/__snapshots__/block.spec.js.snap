// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Faculty Profile Card Block 🔎 Block should be available. 1`] = `
"<!-- wp:purdue-blocks/faculty-profile-card -->
<div class=\\"faculty-profile-card box\\"><div class=\\"media\\"><div class=\\"media-left\\"><div class=\\"image\\"><img src=\\"\\"/></div></div><div class=\\"media-content\\"><div class=\\"content\\"><ul></ul></div></div></div><div class=\\"level is-mobile\\"></div></div>
<!-- /wp:purdue-blocks/faculty-profile-card -->"
`;

exports[`🔬 Faculty Profile Card Block 🔬 Block Editor Fields 🔬 Personal Info Fields 🔎 Add Email Address Field 1`] = `
"<!-- wp:purdue-blocks/faculty-profile-card {\\"email\\":\\"test@test.test\\"} -->
<div class=\\"faculty-profile-card box\\"><div class=\\"media\\"><div class=\\"media-left\\"><div class=\\"image\\"><img src=\\"\\"/></div></div><div class=\\"media-content\\"><div class=\\"content\\"><ul><li><i class=\\"fas fa-envelope\\" aria-hidden=\\"true\\"></i><div class=\\"profile-info-item\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"mailto:test@test.test\\">test@test.test</a><span>Email</span></div></li></ul></div></div></div><div class=\\"level is-mobile\\"></div></div>
<!-- /wp:purdue-blocks/faculty-profile-card -->"
`;

exports[`🔬 Faculty Profile Card Block 🔬 Block Editor Fields 🔬 Personal Info Fields 🔎 Add Personal Website Field 1`] = `
"<!-- wp:purdue-blocks/faculty-profile-card {\\"personalLink\\":\\"https://www.purdue.edu\\"} -->
<div class=\\"faculty-profile-card box\\"><div class=\\"media\\"><div class=\\"media-left\\"><div class=\\"image\\"><img src=\\"\\"/></div></div><div class=\\"media-content\\"><div class=\\"content\\"><ul><li><i class=\\"fas fa-desktop\\" aria-hidden=\\"true\\"></i><div class=\\"profile-info-item\\"><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" href=\\"https://www.purdue.edu\\">https://www.purdue.edu</a><span>Personal Website</span></div></li></ul></div></div></div><div class=\\"level is-mobile\\"></div></div>
<!-- /wp:purdue-blocks/faculty-profile-card -->"
`;

exports[`🔬 Faculty Profile Card Block 🔬 Block Editor Fields 🔬 Personal Info Fields 🔎 Add Phone Number Field 1`] = `
"<!-- wp:purdue-blocks/faculty-profile-card {\\"phone\\":\\"123 456 7890\\"} -->
<div class=\\"faculty-profile-card box\\"><div class=\\"media\\"><div class=\\"media-left\\"><div class=\\"image\\"><img src=\\"\\"/></div></div><div class=\\"media-content\\"><div class=\\"content\\"><ul><li><i class=\\"fas fa-phone\\" aria-hidden=\\"true\\"></i><div class=\\"profile-info-item\\"><p>123 456 7890</p><span>Phone</span></div></li></ul></div></div></div><div class=\\"level is-mobile\\"></div></div>
<!-- /wp:purdue-blocks/faculty-profile-card -->"
`;

exports[`🔬 Faculty Profile Card Block 🔬 Side Panel Settings 🔎 Image alt text field 1`] = `
"<!-- wp:purdue-blocks/faculty-profile-card {\\"altText\\":\\"Image alt text.\\"} -->
<div class=\\"faculty-profile-card box\\"><div class=\\"media\\"><div class=\\"media-left\\"><div class=\\"image\\"><img src=\\"\\"/></div></div><div class=\\"media-content\\"><div class=\\"content\\"><ul></ul></div></div></div><div class=\\"level is-mobile\\"></div></div>
<!-- /wp:purdue-blocks/faculty-profile-card -->"
`;

exports[`🔬 Faculty Profile Card Block 🔬 Side Panel Settings 🔬 Social Share Links 🔎 "Include Social Share Links" checkbox reveals individual checkboxes 1`] = `
"<!-- wp:purdue-blocks/faculty-profile-card {\\"includeSocial\\":true} -->
<div class=\\"faculty-profile-card box\\"><div class=\\"media\\"><div class=\\"media-left\\"><div class=\\"image\\"><img src=\\"\\"/></div></div><div class=\\"media-content\\"><div class=\\"content\\"><ul></ul></div></div></div><div class=\\"level is-mobile\\"><div class=\\"level-right content\\"><p class=\\"level-item\\">Follow Me: </p></div></div></div>
<!-- /wp:purdue-blocks/faculty-profile-card -->"
`;

exports[`🔬 Faculty Profile Card Block 🔬 Side Panel Settings 🔬 Social Share Links 🔎 Social links added correctly 1`] = `
"<!-- wp:purdue-blocks/faculty-profile-card {\\"includeSocial\\":true,\\"checkedSocials\\":{\\"Facebook\\":{\\"slug\\":\\"facebook-square\\",\\"checked\\":true},\\"Twitter\\":{\\"slug\\":\\"twitter-square\\",\\"checked\\":true},\\"LinkedIn\\":{\\"slug\\":\\"linkedin\\",\\"checked\\":true},\\"Instagram\\":{\\"slug\\":\\"instagram\\",\\"checked\\":true}}} -->
<div class=\\"faculty-profile-card box\\"><div class=\\"media\\"><div class=\\"media-left\\"><div class=\\"image\\"><img src=\\"\\"/></div></div><div class=\\"media-content\\"><div class=\\"content\\"><ul></ul></div></div></div><div class=\\"level is-mobile\\"><div class=\\"level-right content\\"><p class=\\"level-item\\">Follow Me: </p><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"level-item\\" href=\\"undefined\\"><i class=\\"fab fa-facebook-square\\" aria-hidden=\\"true\\"></i></a><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"level-item\\" href=\\"undefined\\"><i class=\\"fab fa-twitter-square\\" aria-hidden=\\"true\\"></i></a><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"level-item\\" href=\\"undefined\\"><i class=\\"fab fa-linkedin\\" aria-hidden=\\"true\\"></i></a><a target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"level-item\\" href=\\"undefined\\"><i class=\\"fab fa-instagram\\" aria-hidden=\\"true\\"></i></a></div></div></div>
<!-- /wp:purdue-blocks/faculty-profile-card -->"
`;
